r/R/table.R

```diff
@@ -261,28 +261,38 @@ names.Table <- function(x) x$ColumnNames()
 `[[<-.Table` <- function(x, i, value) {
   if (is.null(value)) {
     if (is.character(i)) {
-      i <- match(i, names(x)) - 1L
+      i <- match(i, names(x))
     }
-    x <- x$RemoveColumn(i)
+    x <- x$RemoveColumn(i - 1L)
   } else {
     if (!is.character(i)) {
       # get or create a/the column name
-      if (i <= ncol(x)) {
-        i <- names(x)[[i]]
+      if (i <= x$num_columns) {
+        i <- names(x)[i]
       } else {
         i <- as.character(i)
       }
     }
 
+    # auto-magic recycling
```

Comment:
1) We should probably only do recycling if `value` is not an ArrowObject (perhaps with a TODO to support Arrow Scalar recycling)
2) https://vctrs.r-lib.org/reference/vec_recycle.html may be a good solution instead of this bespoke code (we already import vctrs so it's available), or else use `base::rep_len` (cf. https://github.com/apache/arrow/blob/master/r/R/array.R#L291)

Reply:
TIL about `vctrs::vec_recycle` that's really nice.